File: StockChess/StockChessCS/Converters/RankRowConverter.cs
using System;
using System.Globalization;
using System.Windows.Data;

namespace StockChessCS.Converters
{
    public class RankRowConverter : IValueConverter
    {
        public object Convert(object value, Type targetType, object parameter, CultureInfo culture)
        {
            var rank = (int) value;
            var row = 8 - rank;
            return row;
        }

        public object ConvertBack(object value, Type targetType, object parameter, CultureInfo culture)
        {
            return Binding.DoNothing;
        }
    }
}

File: StockChess/StockChess/Converters/RankRowConverter.vb
Imports System.Globalization

Public Class RankRowConverter
    Implements IValueConverter

    Public Function Convert(value As Object, targetType As Type, parameter As Object, culture As CultureInfo) As Object Implements IValueConverter.Convert
        Dim rank = CInt(value)
        Dim row = 8 - rank
        Return row
    End Function

    Public Function ConvertBack(value As Object, targetType As Type, parameter As Object, culture As CultureInfo) As Object Implements IValueConverter.ConvertBack
        Return Binding.DoNothing
    End Function
End Class
